### PR TITLE
add more tests for webhook

### LINKED ISSUE #2938: Add more unit tests to cover webhook functionalities
This is part of  #2822 

Current status:
image

https://codecov.io/gh/pingcap/tidb-operator/tree/master/pkg/webhook

### PR DESCRIPTION
**Pull request by cofyc**
### What problem does this PR solve? 
fixes #2938
### Check List 

Tests 

 - Unit test
 - E2E test
 - Stability test
 - Manual test (add detailed scripts or steps below)
 - No code

Code changes

 - Has Go code change
 - Has CI related scripts change
 - Has Terraform scripts change

Side effects

 - Breaking backward compatibility

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation

### Does this PR introduce a user-facing change?:

```release-note
NONE
```

### CODE REVIEW

File: pkg/webhook/pod/pod_mutater.go [hunk 1]
```diff
@@ -57,11 +57,12 @@ func (pc *PodAdmissionControl) mutatePod(ar *admissionv1beta1.AdmissionRequest)
 		return util.ARFail(err)
 	}
 
-	if features.DefaultFeatureGate.Enabled(features.AutoScaling) {
-		err := pc.tikvHotRegionSchedule(tc, pod)
-		if err != nil {
-			return util.ARFail(err)
-		}
+	if !features.DefaultFeatureGate.Enabled(features.AutoScaling) {
```
Comment (DanielZhangQD) [inline pkg/webhook/pod/pod_mutater.go:60]: Should we move this to the beginning of the function?

Comment (cofyc) [reply pkg/webhook/pod/pod_mutater.go:60]: SGTM

Comment (cofyc) [reply pkg/webhook/pod/pod_mutater.go]: done
I also added a new test `TestValidate`

**Pull request merged**